symphony/bdk/core/service/message/message_parser.py

```diff
@@ -0,0 +1,103 @@
+"""This module contains a set of util functions that can be used on incoming V4Message such as:
+extracting entities like Mentions, Hashtags, Cashtags, Emojis and extracting text content from presentationML
+contained in the message.
+"""
+import json
+
+from enum import Enum
+from json import JSONDecodeError
+from xml.etree import ElementTree as ET
+from xml.etree.ElementTree import ElementTree, fromstring, tostring
+
+from symphony.bdk.core.service.exception import MessageParserError
+from symphony.bdk.gen.agent_model.v4_message import V4Message
+
+
+def get_text_content_from_message(message: V4Message):
+    """Get text content from PresentationML on incoming messages
+
+    :param message: message containing the PresentationML to be parsed
+    :return: the message text content extracted from the given PresentationML
+    """
+
+    unparsable_presentationml = "Unable to parse the PresentationML, it is not in the correct format."
+    presentation_ml = message.message
+    try:
+        tree = ElementTree(fromstring(presentation_ml))
+    except ET.ParseError:
+        raise MessageParserError(unparsable_presentationml)
+    return tostring(tree.getroot(), method="text").decode().strip()
+
+
+def get_mentions(message: V4Message) -> [int]:
+    """ Parse data inside an incoming message and returns a list containing the user ids corresponding
+    to the users mentioned
+
+    :param message: incoming V4 message to be parsed
+    :return: list of users ids that has been mentioned inside the message
+    """
+    mentions_list = _get_tags(message, _EntityTypeEnum.MENTION)
+    return [int(user_id) for user_id in mentions_list]
+
+
+def get_hashtags(message: V4Message):
```

Comment:
Please add type hints for the return type.

Reply:
done in the next commit